BrightFutures/ExecutionContext.swift

```diff
@@ -50,9 +50,22 @@ public func toContext(queue: dispatch_queue_t) -> ExecutionContext {
     return Queue(queue: queue).context
 }
 
-typealias ThreadingModel = () -> ExecutionContext
+/// Defines BrightFutures' default threading behaviour, by default this is
+/// set to `StandardThreadingModel.AsynchronousOnCurrentThread`
+public var DefaultThreadingModel: StandardThreadingModel = .AsynchronousOnCurrentThread
```

Comment:
Like I said in my comment, I deliberately didn't do that because it completely removes any safety with the threading models. 

If we do decide to open the default threading models to users then I'll implement that, but there needs to be a discussion about that, and I'd like to know @Thomvis's thoughts first.